### DISCUSSION TITLE
[Question] Duplicate insertion with custom primary key

### DISCUSSION THREAD

**Discussion by amirasyraf** [jenssegers/laravel-mongodb#2117] [Q&A] [answered]
Hi,

Lets say I have the following model:

```
<?php

class Order extends Model
{
    use HasFactory;

    protected $connection = 'mongodb';
    protected $primaryKey = 'customKey';
    
    protected $guarded = [];
}
```
Then,

```
// First insert
$order = Order::create([
    'customKey' => 'key1',
    'data'      => $data,
]);

// Second insert
$order = Order::create([
    'customKey' => 'key1',
    'data'      => $data,
]);
```

The second insert is still successful despite it having the same customKey `key1` which already exists. I thought with the primary key defined, it would throw an error (or ignore) when inserting a duplicate key? Is this not the case?

I don't think `insertOrIgnore` is supported either. Any  alternative solution to this?

**Comment by divine** [comment, marked as answer]
Hello,

Have you created correct index within mongodb itself?

Thanks!

**Reply by amirasyraf** [reply]
I wasn't aware I needed to create the index manually first. But I have tried it just now and I'm still able to insert the duplicate key. 

Thank you.

**Reply by divine** [reply]
This is a database, you should create indexes (or in laravel schemas and import it).

Have you created a [unique index](https://docs.mongodb.com/manual/core/index-unique/#create-a-unique-index) or just plain one?

**Reply by amirasyraf** [reply]
Yep, I did create a unique index:

```
db.mycollection.createIndex({customKey: 1}, {unique: true})
```

Definitely can still insert duplicate keys.

**Reply by divine** [reply]
Double check that you've created correctly it.

If you're still able to create duplicate keys then you're doing something wrong with your database.

_Not related to this library at all._
